xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -107,3 +114,90 @@ async def send_reliable_submission(
     return await _wait_for_final_transaction_outcome(
         transaction_hash, client, prelim_result, transaction.last_ledger_sequence
     )
+
+
+async def _get_signed_tx(
+    transaction: Union[Transaction, str],
+    client: Client,
+    wallet: Optional[Wallet] = None,
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the transaction or transaction blob to be signed if unsigned.
+        client: the network client to autofill from.
+        wallet: an optional wallet with which to sign the transaction. This is
+            only needed if the transaction is unsigned.
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if isinstance(transaction, str):
+        transaction = Transaction.from_blob(transaction)
+
+    if transaction.is_signed():
+        return transaction
+
+    if not wallet:
+        raise XRPLException(
+            "Wallet must be provided when submitting an unsigned transaction"
+        )
```

Comment:
nit: I'd recommend including the else even though it's not strictly necessary because the logic relies on that return to make sense. Each time I've reviewed the PR it's taken me a minute or two to remember why it's guaranteed that at this point the transaction MUST require a signature, and I think adding this else would clarify that.
```suggestion
    if transaction.is_signed():
        return transaction
    else:
      if not wallet:
          raise XRPLException(
              "Wallet must be provided when submitting an unsigned transaction"
          )
```

Reply:
I think adding a comment saying "Attempts to validate, autofill, and sign a transaction when unsigned" would suffice. The `return` statement already tells that we only proceed to the remaining steps for unsigned transaction and I would prefer not to have many conditional layers in the code.